tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/decisiontree/DecisionTree.scala

```diff
@@ -129,17 +130,5 @@ case class DecisionTree(outcomes: Iterable[Int], child: IndexedSeq[Map[Int, Int]
 }
 
 object DecisionTree {
-  // Override the Spray/JSON serialization for maps with integer keys, because these
-  // don't work out-of-the-box.
-  import spray.json.DefaultJsonProtocol.{ mapFormat => _, _ }
-  implicit val intMapFormat = new JsonFormat[Map[Int, Int]] {
-    override def write(map: Map[Int, Int]): JsValue = {
-      map.toSeq.toJson
-    }
-    override def read(json: JsValue): Map[Int, Int] = json match {
-      case value: JsArray => value.convertTo[Seq[(Int, Int)]].toMap
-      case _ => throw new DeserializationException("Expected JSArray for int map")
-    }
-  }
-  implicit val dtFormat = jsonFormat4(DecisionTree.apply)
+  implicit val decisionTreeFormat = jsonFormat4(DecisionTree.apply)
```

Comment:
Yes. I made it serialize `Map[String,T]` as a JSON object, and all other `Map[K,T]` as an array of two-element items (what you were coding by hand before).

`spray-json` instead checked (at runtime) if the serialization of the key type was a `JsString`, and if not, threw an exception. Icky! My version is a bit less readable if you're hand-coding JSON, but it's otherwise nicer. IMHO. :)